### ISSUE TITLE
More compound generate.

### ISSUE THREAD

**Issue by Aunity** [ParmEd/ParmEd#1267]
I'm using parmed to combine gromacs topology files. The number of molecules before combine is 1. But it changed to 2 while I combined it together. Here is my code

```python
import parmed  as pmd
gmxtop = pmd.load_file('MOL_GMX.top')
gmxtop.save('lig.top')
```

Molecules in the topology file before combine:
```plaintext
[ system ]
 MOL

[ molecules ]
; Compound        nmols
 MOL              1
```

Molecules in the topology file after combine:
```plaintext
[ system ]
; Name
Generic title

[ molecules ]
; Compound       #mols
MOL                  2
```
[gromacs.zip](https://github.com/ParmEd/ParmEd/files/9581586/gromacs.zip)

**Comment by swails** [comment]
I can confirm this is a bug, and I know what's happening, but it's really hard to fix without completely destroying performance.

You can fix this with by changing `gmxtop.save('lig.top')` to `gmxtop.save('lig.top', combine='all')`.

The problem is that your MOL definition (from `MOL_GMX.itp`) has three atoms (atom numbers 2, 73, and 74) that are actually a separate water molecule (that is, they are bonded to each other but not to any other atom in the system).  Note that this is actually a water molecule, but whose atom indices are interspersed in the rest of the ligand.

ParmEd detects this and tries to split the system up into molecules to write them out separately, but it uses residue information to identify the individual molecules.  For single-residue molecules, the molecule name is assigned the residue name (in your case, MOL).  However, since both molecules are part of the *same* residue, ParmEd gets confused and thinks there are two of those molecules in the system.

If you look carefully at the lig.top file that is generated, you'll see that, while there are 2 MOL molecules, the MOL definition has only 143 atoms compared to the 146 atoms in `MOL_gmx.itp`.


------

In summary, you can add the `combine='all'` argument to skip the molecule splitting step, but a more robust fix would be to fix your input topology file and separate the water and ligand atoms into separate residues.
